ledger/acctupdates.go

```
@@ -376,60 +523,526 @@ func (au *accountUpdates) getAssetCreatorForRound(rnd basics.Round, aidx basics.
-		assetFlushcount = make(map[basics.AssetIndex]int)
+	au.accountsMu.RUnlock()
+
+	for i := uint64(0); i < offset; i++ {
+		for addr := range deltas[i] {
+			flushcount[addr] = flushcount[addr] + 1
+		}
+	}
+
+	var catchpointLabel string
+	var committedRoundDigest crypto.Digest
+
+	err := au.dbs.wdb.Atomic(func(tx *sql.Tx) (err error) {
+		treeTargetRound := basics.Round(0)
+		if au.catchpointInterval > 0 {
+			mc, err0 := makeMerkleCommitter(tx, false)
+			if err0 != nil {
+				return err0
+			}
+			if au.balancesTrie == nil {
+				trie, err := merkletrie.MakeTrie(mc, trieCachedNodesCount)
+				if err != nil {
+					au.log.Warnf("unable to create merkle trie during committedUpTo: %v", err)
+					return err
+				}
+				au.balancesTrie = trie
+			} else {
+				au.balancesTrie.SetCommitter(mc)
+			}
+			treeTargetRound = dbRound + basics.Round(offset)
+		}
 		for i := uint64(0); i < offset; i++ {
-			rnd := au.dbRound + basics.Round(i) + 1
-			err := accountsNewRound(tx, rnd, au.deltas[i], au.roundTotals[i+1].RewardsLevel, au.protos[i+1])
+			err = accountsNewRound(tx, deltas[i], roundTotals[i+1].RewardsLevel, protos[i+1])
 			if err != nil {
 				return err
 			}
 
-			for aidx := range au.assetDeltas[i] {
-				assetFlushcount[aidx] = assetFlushcount[aidx] + 1
-			}
-
-			for addr := range au.deltas[i] {
-				flushcount[addr] = flushcount[addr] + 1
+			err = au.accountsUpdateBalaces(deltas[i])
+			if err != nil {
+				return err
 			}
 		}
-		return nil
+		err = updateAccountsRound(tx, dbRound+basics.Round(offset), treeTargetRound)
+		return err
 	})
+
 	if err != nil {
+		au.balancesTrie = nil
 		au.log.Warnf("unable to advance account snapshot: %v", err)
-		return au.dbRound
+		return
+	}
+	if isCatchpointRound {
+		committedRoundDigest = au.roundDigest[offset+uint64(lookback)-1]
+		catchpointLabel, err = au.accountsCreateCatchpointLabel(dbRound+basics.Round(offset)+lookback, dbRound+basics.Round(offset), committedRoundDigest)
```

err is not being checked

---

added a logging around that.